### ISSUE TITLE
Unable to complete Moonbeam transactions (Gas estimation fails)

### ISSUE THREAD

**Issue by josheleonard** [brave/brave-browser#23529]
## Description 

Reported via Brave community: https://community.brave.com/t/cannot-transfer-glmr-token-out-from-my-brave-wallet/379404

## Steps to Reproduce

   1. Create a Brave Wallet 
   2. Add Moonbeam via chainlist.org
   3. Fund account with GLMR - faucet: https://glmr.supply/
   4. Attempt to send GLMR

## Actual result:

Confirmation panel never opens, console logs `err=Failed to get the gas fees for EIP-1559 transaction`

## Expected result:
Confirmation panel opens, transaction completes after signing

## Reproduces how often: 
Easily reproduced

## Brave version (brave://version info)

1.39.120 Chromium: 102.0.5005.99 (Official Build) (arm64) 
870f7bcc58dfa811cc68c2186439721385e086d0-refs/branch-heads/5005@{#1125}
OS | macOS Version 12.4 (Build 21F79)

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome?

**Comment by bbondy** [comment]
@onyb what do you think is the cause of this one?

**Comment by onyb** [comment]
It seems the `blockCount` parameter of `eth_feeHistory` no longer accepts an integer, and expects a hex-encoded string. See docs: https://ethereum.github.io/execution-apis/api-documentation

geth supports both integer and hex strings, but apparently Moonbeam does not:
```
curl -s https://rpc.ankr.com/moonbeam -X POST -H "Content-Type: application/json" -d '{"id":1,"jsonrpc":"2.0","method":"eth_feeHistory","params":[300, "latest", [20,50,80]]}' | jq      
{
  "jsonrpc": "2.0",
  "error": {
    "code": -32602,
    "message": "Invalid params: invalid type: integer `300`, expected a (both 0x-prefixed or not) hex string with length between (0; 64]."
  },
  "id": 1
}
```

Changing `blockCount` parameter to use 0x-prefixed hex string fixed the issue. I suggest we apply this change without making an exception for Moonbeam. @bbondy

**Comment by onyb** [comment]
Related geth fix here: https://github.com/ethereum/go-ethereum/pull/23239, included in `1.10.7`.
